equational_theories/EquationalResult.lean

```diff
@@ -0,0 +1,103 @@
+import Lean.Elab.Exception
+import Lean.Elab.Declaration
+
+import equational_theories.ParseImplications
+
+open Lean Parser Elab Command
+
+namespace Result
+
+/-- An entry in the equational results environment extension.
+-/
+inductive EntryVariant where
+  | implication : Implication → EntryVariant
+  | nonimplication : Implication → EntryVariant
+  /-- An equation that always holds. -/
+  | unconditional : String → EntryVariant
```

Comment:
These two are just instance of the general “∃ magma G such that equations {1,2,3} hold and {4,5,6} do not hold”. Note that that already entails 9 non-implicaitons. Once we add more exhaustive analysis of example magmas (as in #19), listing all non-implications here will not scale, and we’ll probably have to switch to the more compact representation.

Doesn't have to happen in this PR, of course; this is just a heads up.

Reply:
I agree that we will want to add other ways to represent results. That can happen later as needed.

I don't see how "EquationN holds for all magmas" is an instance of "∃ magma G such that equations {1,2,3} hold and {4,5,6} do not hold".

Reply:
A, I misread the comment. My bad